src/tools/Registration/RegistrationUtilities.psm1

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE.txt in the project root for license information.
+
+<#
+
+This module contains utility functions for working with registration resources
+#>
+
+
+<#
+
+.SYNOPSIS
+
+Uses the current Azure Powershell context to retrieve registration resources in Azure from the default resource group
+and with the default resource name (if $AzureStackStampCloudId is provided)
+
+#>
+function Get-AzureRegistrationResource{
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory = $false)]
+    [String] $AzureStackStampCloudId,
+
+    [Parameter(Mandatory = $false)]
+    [String] $ResourceGroupName = "AzureStack",
+
+    [Parameter(Mandatory = $false)]
+    [String] $ResourceName = "AzureStack"
+)
+
+    $VerbosePreference     = "Continue"
+    $ErrorActionPreference = "Stop"
+
+    Write-Verbose "Searching for registration resource using the provided parameters"
+    $registrationResources = Find-AzureRmResource -ResourceNameContains $ResourceName -ResourceType 'Microsoft.AzureStack/registrations' -ResourceGroupNameEquals $ResourceGroupName
+    $registrations = @()
+    foreach ($resource in $registrationResources)
+    {
+        $resource = Get-AzureRmResource -ResourceId $resource.ResourceId
+        if($AzureStackStampCloudId)
+        {
+            if ($resource.Properties.CloudId -eq $AzureStackStampCloudId)
+            {
+                Write-Verbose "Registration resource found:`r`n$(ConvertTo-Json $resource)"
+                return $resource
+            }
+        }
+        else
+        {
+            $registrations += $resource
+        }
+    }
+
+    if ($registrations.Count -gt 0)
+    {
+        Write-Verbose "Registrations: $registrations"
+    }
+    else
+    {
+        Write-Verbose "Registration resource(s) could not be located with the provided parameters."
+    }
+}
+
+
+<#
+
+.SYNOPSIS
+
+If the context is set to the Azure Stack environment administrator this will retrieve the activation record in the Azure Stack
+if it has been created via successful registration run. 
+
+#>
+function Get-AzureStackActivationRecord{
+
+    $currentContext = Get-AzureRmContext
```

Comment:
Some modules are using Az & some are using AzureRm. Should we move all modules to use Az?